controllers/scaledobject_controller.go

```diff
@@ -51,7 +51,7 @@ type ScaledObjectReconciler struct {
 	GlobalHTTPTimeout time.Duration
 	Recorder          record.EventRecorder
 
-	scaleClient              *scale.ScalesGetter
```

Comment:
Any particular reason for this change?

Reply:
There are two main reasons:
1. The scaleClient isn't really optional - it's always initialised directly in the `SetupWithManager` func.
2. `scale.ScalesGetter` is an interface and pointers to interfaces are generally discouraged in go - see [here](https://stackoverflow.com/questions/44370277/type-is-pointer-to-interface-not-interface-confusion) for a more in depth explanation. We felt it was a good clean-up.

For full disclosure having pointers to interfaces makes it more difficult to mock them. When creating mocks for our tests this was highlighted and so led us to make the change. We feel the above reasons are more compelling and we generally wouldn't change functional code based on testing requirements.

Reply:
Makes completely sense